common/src/main/scala/org/apache/celeborn/common/rpc/netty/Dispatcher.scala

```diff
@@ -154,18 +155,21 @@ private[celeborn] class Dispatcher(nettyEnv: NettyRpcEnv) extends Logging {
       endpointName: String,
       message: InboxMessage,
       callbackIfStopped: Exception => Unit): Unit = {
+    var data: EndpointData = null
     val error = synchronized {
-      val data = endpoints.get(endpointName)
+      data = endpoints.get(endpointName)
       if (stopped) {
         Some(new RpcEnvStoppedException())
       } else if (data == null) {
         Some(new CelebornException(s"Could not find $endpointName."))
       } else {
-        data.inbox.post(message)
-        receivers.offer(data)
         None
       }
     }
+    if (error.isEmpty) {
```

Comment:
Consider this case:
1. At time 1 thread1 calls `postMessage` and `stopped` is false, then it reaches to 169 and tries to post the message;
2. At time 2 thread2 calls `stop` and stops the dispatcher
3. At time 3 thread1 still posts the message and offers to `receivers`

Prior to this PR the above will not happen because it's always inside the synchronize block

Reply:
you're right, updated here